ratpack-core/src/main/java/ratpack/http/client/internal/NoopFixedChannelPoolHandler.java

```diff
@@ -29,21 +32,26 @@
 
   private final String host;
   private final Duration idleTimeout;
+  private final ProxyInternal proxy;
 
-  public NoopFixedChannelPoolHandler(HttpChannelKey channelKey, Duration idleTimeout) {
+  public NoopFixedChannelPoolHandler(HttpChannelKey channelKey, Duration idleTimeout, ProxyInternal proxy) {
     this.host = channelKey.host;
     this.idleTimeout = idleTimeout;
+    this.proxy = proxy;
   }
 
   @Override
   public void channelCreated(Channel ch) throws Exception {
+    if  (proxy != null && proxy.shouldProxy(host)) {
+      SocketAddress proxyAddress = new InetSocketAddress(proxy.getHost(), proxy.getPort());
+      ch.pipeline().addLast(new HttpProxyHandler(proxyAddress));
+    }
   }
 
   @Override
   public void channelReleased(Channel ch) throws Exception {
     if (ch.isOpen()) {
       ch.config().setAutoRead(true);
-      ch.pipeline().addLast(IdlingConnectionHandler.INSTANCE);
```

Comment:
@drmaas was this intentional or did one of us (probably me) keyboard slip here?

Reply:
@johnrengelman looks like it was you -  this is needed so that a channel released back into the pool gets closed if someone accidentally uses it.